File: mobile/src/main/java/com/ataulm/whatsnext/BaseActivity.java
package com.ataulm.whatsnext;

import android.support.v7.app.AppCompatActivity;

import com.ataulm.support.Clock;

public abstract class BaseActivity extends AppCompatActivity {

    protected WhatsNextService whatsNextService() {
        return ((WhatsNextApplication) getApplication()).whatsNextService();
    }

    protected Clock clock() {
        return ((WhatsNextApplication) getApplication()).clock();
    }

    protected Navigator navigator() {
        return ((WhatsNextApplication) getApplication()).navigator();
    }
}

File: mobile/src/main/java/com/ataulm/whatsnext/BaseActivity.kt
package com.ataulm.whatsnext

import android.support.v7.app.AppCompatActivity

import com.ataulm.support.Clock

abstract class BaseActivity : AppCompatActivity() {

    protected fun whatsNextService(): WhatsNextService {
        return (application as WhatsNextApplication).whatsNextService()
    }

    protected fun clock(): Clock {
        return (application as WhatsNextApplication).clock()
    }

    protected fun navigator(): Navigator {
        return (application as WhatsNextApplication).navigator()
    }
}
